### ISSUE TITLE
DemuxFastqs's quality format is incorrect

### ISSUE THREAD

**Issue by gshiba** [fulcrumgenomics/fgbio#327]
Two issues:
* Auto detection of quality format is not working
* Quality encoding flag `-q` is ignored

Set up:

```
$ zcat r1.fastq.gz
@M00001:001:000000000-AAAAA:1:1101:16729:1382 1:N:0:1
TCGTGACAGGAATCAAATGAAAACACTTGGT
+
1>1>11AFF?FFGGGGGGCBDGBGCFGH11B

$ zcat r2.fastq.gz
@M00001:001:000000000-AAAAA:1:1101:16729:1382 2:N:0:1
TCGTGACAGGAATCAAATGAAAACACTTGGT
+
1>1>11AFF?FFGGGGGGCBDGBGCFGH11B

$ cat samples.csv
Sample_Barcode,Sample_ID,Sample_Name,Library_ID,Description
TCGTTCGT,foo,S1,,

$ fgbio -v
Version: 0.4.0
```

Tool will automatically `Auto-detected quality format as: Standard`, but the output quality is not the same format as input.

```
$ fgbio DemuxFastqs -i r1.fastq.gz r2.fastq.gz -o demuxed --read-structures 4B+T 4B+T --metadata samples.csv
[2017/11/24 16:29:49 | DemuxFastqs | Info] Assuming input metadata file is simple CSV file.
[2017/11/24 16:29:49 | FgBioMain | Info] Executing DemuxFastqs from fgbio version 0.4.0 as gshiba@neptune on JRE 1.8.0_121-b15 with snappy
[2017/11/24 16:29:49 | DemuxFastqs | Warning] Making ambiguous determination about fastq's quality encoding; possible encodings: Standard, Solexa, Illumina.
[2017/11/24 16:29:49 | DemuxFastqs | Info] Auto-detected quality format as: Standard
[2017/11/24 16:29:50 | FgBioMain | Info] DemuxFastqs completed. Elapsed time: 0.08 minutes.

$ ll demuxed/
demux_barcode_metrics.txt  foo-S1-TCGTTCGT.bam        unmatched.bam

$ samtools view demuxed/foo-S1-TCGTTCGT.bam
M00001:001:000000000-AAAAA:1:1101:16729:1382    77      *       0       0       *       *       0       0       GACAGGAATCAAATGAAAACACTTGGT     RRbgg`gghhhhhhdcehchdghiRRc     BC:Z:TCGT-TCGT  RG:Z:foo
M00001:001:000000000-AAAAA:1:1101:16729:1382    141     *       0       0       *       *       0       0       GACAGGAATCAAATGAAAACACTTGGT     RRbgg`gghhhhhhdcehchdghiRRc     BC:Z:TCGT-TCGT  RG:Z:foo
```

I can't override the quality encoding, even with the `-q` flag:

```
$ fgbio DemuxFastqs -i r1.fastq.gz r2.fastq.gz -o demuxed --read-structures 4B+T 4B+T --metadata samples.csv  -q Illumina
[2017/11/24 16:30:36 | DemuxFastqs | Info] Assuming input metadata file is simple CSV file.
[2017/11/24 16:30:36 | FgBioMain | Info] Executing DemuxFastqs from fgbio version 0.4.0 as gshiba@neptune on JRE 1.8.0_121-b15 with snappy
[2017/11/24 16:30:36 | DemuxFastqs | Warning] Making ambiguous determination about fastq's quality encoding; possible encodings: Standard, Solexa, Illumina.
[2017/11/24 16:30:36 | DemuxFastqs | Info] Auto-detected quality format as: Standard
[2017/11/24 16:30:37 | FgBioMain | Info] DemuxFastqs completed. Elapsed time: 0.08 minutes.

$ samtools view demuxed/foo-S1-TCGTTCGT.bam
M00001:001:000000000-AAAAA:1:1101:16729:1382    77      *       0       0       *       *       0       0       GACAGGAATCAAATGAAAACACTTGGT     RRbgg`gghhhhhhdcehchdghiRRc     BC:Z:TCGT-TCGT  RG:Z:foo
M00001:001:000000000-AAAAA:1:1101:16729:1382    141     *       0       0       *       *       0       0       GACAGGAATCAAATGAAAACACTTGGT     RRbgg`gghhhhhhdcehchdghiRRc     BC:Z:TCGT-TCGT  RG:Z:foo
```

(I actually don't know that my fastqs quality format is *called*. It's from a MiSeq.)

Thanks

**Comment by nh13** [comment]
@gshiba thank-you for the very easy to follow debugging example.  You're absolutely right, the `--quality-format` option was not being respected AND the output base qualities were wrong when outputting to BAM.  My apologies for the confusion, and thank-you for spotting the bug.  I have a potential fix in #329.  I will update here the progress on getting the fix merged.

**Comment by nh13** [comment]
@gshiba this issue has been fixed.
